sklearn/linear_model/_glm/glm.py

```diff
@@ -263,13 +264,29 @@ def fit(self, X, y, sample_weight=None):
                 coef = self.coef_
             coef = coef.astype(loss_dtype, copy=False)
         else:
+            # Note: zero init for both the coefficients and the intercept
+            # is needed if we want the unpenalized model to converge to the
+            # minimum nornm solution which is likely to generalize better
+            # than other solutions.
```

Comment:
```suggestion
            # Note: Instead of smarter starting values, we initialize both
            # the coefficients and the intercept with zeros. This is needed
            # if we want the unpenalized model to converge to the
            # minimum norm solution which is likely to generalize better
            # than other solutions.
```